.empty-mod/go.mod

```diff
@@ -0,0 +1,11 @@
+// module empty-mod is an empty module that's used to help with containerd
+// having a circular dependency on itself through plugin modules.
+//
+// We use this module as a "replace" rule in containerd's go.mod, to prevent
+// relying on transitive dependencies (coming from older versions of containerd
+// defined on plugin go.mod).
+//
+// The replace rule forces go modules to consider the "current" version of
+// containerd to be the source of truth, helping us catch missing go.mod rules,
+// or version changes early.
+module empty-mod
```

Comment:
interesting technique! has this been used anywhere/anybody else?

Reply:
My original attempt was to use `replace github.com/containerd/containerd => ./`, but that didn't work due to limitations of go modules (https://github.com/golang/go/issues/45492); the "empty module" approach was discussed in https://github.com/golang/go/issues/34417, which looks like other people also used it to fix circular dependencies.

I think go modules should be able to handle this automatically, not requiring any of these tricks; perhaps I should write up a features request / proposal for that in the Golang issue tracker.

Reply:
thanks for the additional details @thaJeztah